packages/engine/src/avatar/systems/AvatarAnimationSystem.tsx

```diff
@@ -338,9 +348,23 @@ const Reactor = () => {
   return null
 }
 
+/**
+ * @todo replace this with a retargeting utility to retarget the source animation assets rather than every time on load,
+ * and introduce a loader function that only loads the necessary data to avoid cleanup of the ecs armature
+ */
+export const setupMixamoAnimation = (entity: Entity) => {
+  normalizeAnimationClips(entity)
+  setComponent(entity, AvatarRigComponent)
+  traverseEntityNode(entity, (child) => {
+    const name = getComponent(child, NameComponent).replace(':', '')
+    if (mixamoVRMRigMap[name]) AvatarRigComponent.setBone(entity, child, mixamoVRMRigMap[name])
```

Comment:
ah, i see this is where we are using setBone. is there a reason why we have one place setting this data internally and one externally?

Reply:
Typically the vrm bone map is created on conversion to VRM. This is the only time it happens for any actual avatar. This reactor handles Imported animations, which are a special case. We have this avatar rig that needs to be mapped to VRM but there's no reason to create a VRM from the rig. To prevent the extra logic being run, I opted to break out the only necessary processing that needs to happen.